### ISSUE TITLE
WebDAV support 

### ISSUE THREAD

**Issue by MrSuicideParrot** [fleetingnotes/fleeting-notes-flutter#654]
For a long time, I've been looking for a self-host replacement of Google Keep. The majority of the applications fail in their design or save the notes in a custom format, which are no goes for anyone looking for a replacement. 

Your app doesn't fail in any of these points, your design is awesome, very similar to Google Keep, and saves the notes in markdown. 

I know the objective of this app is to easily ingest notes to Obsidian on mobile phone using obsidian sync. But would you consider to add WebDAV support? This would work similar to local file sync.

Advantages:
 
People could save their notes in WebDAV servers (ex. Nextcloud). On the other hand, we would be able to use this app in a standalone way.

**Comment by matthewwong525** [comment]
Is webdav done through an API call? If so, it may be supportable through the plugin system with the https://github.com/fleetingnotes/fleeting-notes-plugins/issues/7

**Comment by MrSuicideParrot** [comment]
Yes, it is a an API call. But if it is a one way sync, I wouldn't be able to synchronize multiple clients through a plugin, right?
